src/services/shopify-client.test.ts

```diff
@@ -0,0 +1,162 @@
+  it('should set connected shop without list correctly', async () => {
+    // Arrange
+    const request = jest.fn(async () => connectedWithoutListResponse);
+
+    const shopifyClient = createHttpShopifyClient({ request });
+
+    // Act
+    const result = await shopifyClient.getShopifyData();
+
+    // Assert
+    expect(request).toBeCalledTimes(1);
+    expect(result).not.toBe(undefined);
+    expect(result.success).toBe(true);
+    expect(result.value[0].shopName).toEqual(connectedWithoutListResponse.data[0].shopName);
+    expect(result.value[0].list.state).toEqual(SubscriberListState.notAvailable);
+  });
+
+  it('should set connected shop with list connected correctly', async () => {
+    // Arrange
+    const request = jest.fn(async () => connectedWithListReadyResponse);
+
+    const shopifyClient = createHttpShopifyClient({ request });
+
+    // Act
+    const result = await shopifyClient.getShopifyData();
+
+    // Assert
+    expect(request).toBeCalledTimes(1);
+    expect(result).not.toBe(undefined);
+    expect(result.success).toBe(true);
+    expect(result.value[0].shopName).toEqual(connectedWithListReadyResponse.data[0].shopName);
+    expect(result.value[0].list.name).not.toBe(undefined);
+    expect(result.value[0].list.amountSubscribers).not.toBe(undefined);
+    expect(result.value[0].list.state).toEqual(SubscriberListState.ready);
+  });
+
+  it('should set connected shop with list connected but synchronizing contacts', async () => {
+    // Arrange
+    const request = jest.fn(async () => connectedWithListSyncResponse);
+
+    const shopifyClient = createHttpShopifyClient({ request });
+
+    // Act
+    const result = await shopifyClient.getShopifyData();
+
+    // Assert
+    expect(request).toBeCalledTimes(1);
+    expect(result).not.toBe(undefined);
+    expect(result.success).toBe(true);
+    expect(result.value[0].shopName).toEqual(connectedWithListReadyResponse.data[0].shopName);
+    expect(result.value[0].list.name).not.toBe(undefined);
+    expect(result.value[0].list.amountSubscribers).not.toBe(undefined);
+    expect(result.value[0].list.state).toEqual(SubscriberListState.synchronizingContacts);
+  });
+
+  it('should set not connected shop', async () => {
+    // Arrange
+    const request = jest.fn(async () => notConnected);
+
+    const shopifyClient = createHttpShopifyClient({ request });
+
+    // Act
+    const result = await shopifyClient.getShopifyData();
+
+    // Assert
+    expect(request).toBeCalledTimes(1);
+    expect(result).not.toBe(undefined);
+    expect(result.success).toBe(true);
+    expect(result.value).toEqual([]);
+  });
+
+  it('should set error connecting shopify', async () => {
+    // Arrange
+    const request = jest.fn(async () => {});
+
+    request.mockImplementation(() => {
+      throw new Error();
+    });
+
+    const shopifyClient = createHttpShopifyClient({ request });
+    console.error = jest.fn(); // silence console error for this test run only
```

Comment:
😮

Reply:
Are you sure that this is not affecting other tests?

Reply:
In the real service when it throws an error, the console shows the error, here it's very ugly to get the console for tests with not needed messages (remember the act warnings)

Reply:
I know, but I think that this change is not being cleaned. Is it?

Reply:
I will test this issue further to not get side effects here. Thanks.